airflow/utils/operator_resources.py

```diff
@@ -50,6 +50,8 @@ def __init__(self, name, units_str, qty):
         self._qty = qty
 
     def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
```

Comment:
Probably better to use `NotImplemented` instead. https://docs.python.org/3.6/library/constants.html#NotImplemented

Reply:
Great idea. thanks for the comment. updated.